Tests/ValetIntegrationTests/ValetIntegrationTests.swift

```diff
@@ -481,6 +482,57 @@ class ValetIntegrationTests: XCTestCase
             }
         }
     }
+
+    // MARK: codable(forKey:)
+
+    func test_codableForKey_succeedsForValidKey() throws {
+        try allPermutations.forEach { valet in
+            try valet.setCodable(stringArray, forKey: key)
+            XCTAssertEqual(stringArray, try valet.codable(forKey: key))
+        }
+    }
+
+    func test_codableForKey_throwsDecoderErrorOnInvalidType() throws {
+        try allPermutations.forEach { valet in
+            try valet.setCodable(stringArray, forKey: key)
+            XCTAssertThrowsError(try valet.codable(forKey: key) as [Int]) { error in
+                switch error {
+                case DecodingError.typeMismatch:
+                    break // expected
+
+                default:
+                    XCTFail("Unexpected error: \(error)")
+                }
+            }
+        }
+    }
+
+    // MARK: setCodable(:forKey:)
+
+    func test_setCodableForKey_throwsEmptyKeyOnInvalidKey() throws {
+        try allPermutations.forEach { valet in
+            XCTAssertThrowsError(try valet.setCodable(stringArray, forKey: "")) { error in
+                XCTAssertEqual(error as? KeychainError, .emptyKey)
+            }
+        }
+    }
+
+    func test_setCodableForKey_customCoders() throws {
```

Comment:
Let's elaborate about what we're testing with `customCoders`. Maybe (and feel free to correct if this doesn't adequately capture your intent):
```suggestion
    func test_setCodableForKey_utilizesPassedInCoder() throws {
```